src/ring_theory/jacobson_ideal.lean

```diff
@@ -86,6 +97,110 @@ theorem mem_jacobson_iff {I : ideal R} {x : R} :
         neg_mul_eq_neg_mul_symm, neg_mul_eq_mul_neg, mul_comm x y]; exact M.mul_mem_right hy)
     (him hz)⟩
 
+/-- An ideal equals its jacobson radical iff it is the intersection of a set of maximal ideal.
+Allowing the set to include ⊤ is equivalent, and is included only to simplify some proofs. -/
+theorem eq_jacobson_iff_Inf_maximal {I : ideal R} :
+  I.jacobson = I ↔ ∃ M : set (ideal R), (∀ J ∈ M, is_maximal J ∨ J = ⊤) ∧ I = Inf M :=
+begin
+  use λ hI, ⟨{J : ideal R | I ≤ J ∧ J.is_maximal}, ⟨λ _ hJ, or.inl hJ.right, hI.symm⟩⟩,
+  rintros ⟨M, hM, hInf⟩,
+  refine le_antisymm (λ x hx, _) le_jacobson,
+  rw [hInf, mem_Inf],
+  intros I hI,
+  cases hM I hI with is_max is_top,
+  { exact (mem_Inf.1 hx) ⟨le_Inf_iff.1 (le_of_eq hInf) I hI, is_max⟩ },
+  { exact is_top.symm ▸ submodule.mem_top }
+end
+
+/-- An ideal `I` equals its jacobson radical if and only if every element outside `I`
+also lies outside of a maximal ideal containing `I`. -/
+lemma eq_jacobson_iff_not_mem {I : ideal R} :
+  I.jacobson = I ↔ ∀ x ∉ I, ∃ M : ideal R, (I ≤ M ∧ M.is_maximal) ∧ x ∉ M :=
+begin
+  split,
+  { intros h x hx,
+    erw [← h, mem_Inf] at hx,
+    push_neg at hx,
+    exact hx },
+  { refine λ h, le_antisymm (λ x hx, _) le_jacobson,
+    contrapose hx,
+    erw mem_Inf,
+    push_neg,
+    exact h x hx }
+end
+
+theorem map_jacobson_of_surjective {f : R →+* S} (hf : function.surjective f) {I : ideal R} :
+  ring_hom.ker f ≤ I → map f (I.jacobson) = (map f I).jacobson :=
+begin
+  intro h,
+  rw [ideal.jacobson, ideal.jacobson],
```

Comment:
I would use `unfold` because it indicates the intention of working with the definitions:
```suggestion
  unfold ideal.jacobson,
```